HPXMLtoOpenStudio/tests/test_validation.rb

```diff
@@ -0,0 +1,618 @@
+      ['/HPXML/Building/BuildingDetails/Systems/Photovoltaics/PVSystem/ArrayAzimuth'] => 'Expected 1 element(s) for xpath: /HPXML/Building/BuildingDetails/Systems/Photovoltaics/PVSystem/ArrayAzimuth',
+      ['/HPXML/Building/BuildingDetails/Systems/Photovoltaics/PVSystem/ArrayTilt'] => 'Expected 1 element(s) for xpath: /HPXML/Building/BuildingDetails/Systems/Photovoltaics/PVSystem/ArrayTilt',
+      ['/HPXML/Building/BuildingDetails/Systems/Photovoltaics/PVSystem/MaxPowerOutput'] => 'Expected 1 element(s) for xpath: /HPXML/Building/BuildingDetails/Systems/Photovoltaics/PVSystem/MaxPowerOutput',
+      ['/HPXML/Building/BuildingDetails/Appliances/ClothesWasher/SystemIdentifier'] => 'Expected 1 element(s) for xpath: /HPXML/Building/BuildingDetails/Appliances/ClothesWasher/SystemIdentifier',
+      ['/HPXML/Building/BuildingDetails/Appliances/ClothesWasher/IntegratedModifiedEnergyFactor'] => 'Expected 0 or 7 element(s) for xpath: /HPXML/Building/BuildingDetails/Appliances/ClothesWasher/ModifiedEnergyFactor | IntegratedModifiedEnergyFactor | RatedAnnualkWh | LabelElectricRate | LabelGasRate | LabelAnnualGasCost | LabelUsage | Capacity',
+      ['/HPXML/Building/BuildingDetails/Appliances/ClothesWasher'] => 'Expected 1 element(s) for xpath: /HPXML/Building/BuildingDetails/Appliances/ClothesDryer/../ClothesWasher',
+      ['/HPXML/Building/BuildingDetails/Appliances/ClothesDryer/SystemIdentifier'] => 'Expected 1 element(s) for xpath: /HPXML/Building/BuildingDetails/Appliances/ClothesDryer/SystemIdentifier',
+      ['/HPXML/Building/BuildingDetails/Appliances/ClothesDryer/FuelType[text()="natural gas" or text()="fuel oil" or text()="propane" or text()="electricity" or text()="wood" or text()="wood pellets"]'] => 'Expected 1 element(s) for xpath: /HPXML/Building/BuildingDetails/Appliances/ClothesDryer/FuelType[text()="natural gas" or text()="fuel oil" or text()="fuel oil 1" or text()="fuel oil 2" or text()="fuel oil 4" or text()="fuel oil 5/6" or text()="diesel" or text()="propane" or text()="kerosene" or text()="coal" or text()="coke" or text()="bituminous coal" or text()="anthracite coal" or text()="electricity" or text()="wood" or text()="wood pellets"]',
+      ['/HPXML/Building/BuildingDetails/Appliances/Dishwasher/SystemIdentifier'] => 'Expected 1 element(s) for xpath: /HPXML/Building/BuildingDetails/Appliances/Dishwasher/SystemIdentifier',
+      ['/HPXML/Building/BuildingDetails/Appliances/Dishwasher/RatedAnnualkWh'] => 'Expected 0 or 6 element(s) for xpath: /HPXML/Building/BuildingDetails/Appliances/Dishwasher/RatedAnnualkWh | EnergyFactor | LabelElectricRate | LabelGasRate | LabelAnnualGasCost | LabelUsage | PlaceSettingCapacity',
```

Comment:
Why are there some rows like this that include "Expected 0", which implies they are optional, but they are located in the expected_error_msgs hash instead of expected_error_msgs_optional?

Reply:
Sorry, I think the hash name was misleading. My intention here was to test the validation by deleting the elements of the keys. For example, in Dishwasher case, deleting `RatedAnnualkWh `element is expected to generate an error message because only five elements remain.